File: src/go_class_android/app/src/main/java/im/zego/goclass/tool/PermissionHelper.java
package im.zego.goclass.tool;

import androidx.fragment.app.FragmentActivity;

import com.permissionx.guolindev.PermissionX;
import com.permissionx.guolindev.callback.RequestCallback;

public class PermissionHelper {
    private static void onPermissionGranted(FragmentActivity activity, String permission, GrantResult grantResult) {
        PermissionX.init(activity).permissions(new String[]{permission}).request((RequestCallback) (
                (allGranted, grantedList, deniedList) -> {
                    if (allGranted) {
                        grantResult.onGrantResult(true);
                    } else {
                        grantResult.onGrantResult(false);
                    }

                }));
    }

    public static void onAudioPermissionGranted(FragmentActivity activity, GrantResult grantResult) {
        onPermissionGranted(activity, "android.permission.RECORD_AUDIO", grantResult);
    }

    public static void onCameraPermissionGranted(FragmentActivity activity, GrantResult grantResult) {
        onPermissionGranted(activity, "android.permission.CAMERA", grantResult);
    }

    public static void onReadSDCardPermissionGranted(FragmentActivity activity, GrantResult grantResult) {
        onPermissionGranted(activity, "android.permission.READ_EXTERNAL_STORAGE", grantResult);
    }

    public static void onWriteSDCardPermissionGranted(FragmentActivity activity, GrantResult
            grantResult) {
        onPermissionGranted(activity, "android.permission.WRITE_EXTERNAL_STORAGE", grantResult);
    }

    public interface GrantResult {
        void onGrantResult(boolean allGranted);
    }
}

File: src/go_class_android/app/src/main/java/im/zego/goclass/tool/PermissionHelper.kt
package im.zego.goclass.tool

import android.app.Activity
import com.easypermission.*

class PermissionHelper {
    companion object {
        private fun onPermissionGranted(
            activity: Activity,
            permission: String,
            grantResult: (Boolean) -> Unit
        ) {
            if (EasyPermission.isPermissionGrant(activity, permission)) {
                grantResult.invoke(true)
            } else {
                EasyPermission.with(activity)
                    .addPermissions(permission)
                    .addRequestPermissionRationaleHandler(permission)
                    { permission, _, nextAction ->
                        nextAction.next(NextActionType.NEXT)
                    }
                    .request(object : PermissionRequestListener() {
                        override fun onCancel(stopPermission: String) {
                        }

                        override fun onGrant(result: MutableMap<String, GrantResult>) {
                            if (result[permission] == GrantResult.GRANT) {
                                grantResult.invoke(true)
                            } else {
                                grantResult.invoke(false)
                            }
                        }
                    })
            }
        }

        fun onAudioPermissionGranted(
            activity: Activity,
            grantResult: (Boolean) -> Unit
        ) {
            onPermissionGranted(activity, Permission.RECORD_AUDIO, grantResult)
        }

        fun onCameraPermissionGranted(
            activity: Activity,
            grantResult: (Boolean) -> Unit
        ) {
            onPermissionGranted(activity, Permission.CAMERA, grantResult)
        }

        fun onReadSDCardPermissionGranted(
            activity: Activity,
            grantResult: (Boolean) -> Unit
        ) {
            onPermissionGranted(activity, Permission.READ_EXTERNAL_STORAGE, grantResult)
        }

        fun onWriteSDCardPermissionGranted(
            activity: Activity,
            grantResult: (Boolean) -> Unit
        ) {
            onPermissionGranted(activity, Permission.WRITE_EXTERNAL_STORAGE, grantResult)
        }
    }
}